src/mobile/src/main/java/tupperdate/android/home/Home.kt

```diff
@@ -1,98 +1,146 @@
+    val presentRecipe = (RecipeApi.Recipe(
+        "Red lobster",
+        "Look at me, am I not yummy ?",
+        "lobster.jpg"
+    ))
 
-@Composable
-private fun HomeDisconnected(
-    onButtonClick: () -> Unit,
-    modifier: Modifier,
-) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there.")
-        Button(onClick = onButtonClick) {
-            Text("Go to onboarding")
-        }
-    }
+    mainTopBar(onChatClick = onChatClick, onProfileClick = onProfileClick)
+
+    DisplayRecipeCard(
+        presentRecipe = presentRecipe,
+        onLike = { recipeApi.like(presentRecipe) },
+        onDislike = { recipeApi.dislike(presentRecipe) },
+        onReturnClick = onReturnClick,
+        onRecipeClick = onRecipeClick
+    )
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there ${user.displayName}")
-        Button(onClick = onButtonClick) {
-            Text("View Branding")
+    Row(Modifier.fillMaxSize()) {
+        Column(
+            Modifier.fillMaxWidth().align(Alignment.Bottom)
+                .padding(bottom = veryLittleButtonSize.dp)
+        ) {
+            WithConstraints { -> //why no parameter here ??
+
+                //determine height and width of card according to constraints
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
+
+                //the align(..) doesn't work anymore (because of reasons), so we need to
+                //set manually a default pos
+                val defaultPos = ((with(DensityAmbient.current) { constraints.maxWidth.toDp() / 2 })
+                        - boxWidth / 2).value.toInt()
+                val (pos, setPos) = remember { mutableStateOf((defaultPos)) }
+
+                //the defaultPos needs a delta measured
+                val (delta, setDelta) = remember { mutableStateOf(0) }
+                val swipeMargin = (boxWidth).value.toInt()
+
+                Box(modifier = Modifier.align(Alignment.CenterHorizontally)
+                    .width((boxWidth)).height((boxHeight))
+                    .padding(bottom = (veryLittleButtonSize * 0.8f).dp)
+                    .draggable(Orientation.Horizontal,
+                        onDragStarted = {}, onDrag =
+                        { deltaTemp ->
+                            run {
+                                setPos(pos + (deltaTemp * 0.75).toInt())
```

Comment:
Why not calculate pos as `defaultPos + delta * 0.75` and use that wherever you need it ? You're duplicating state.